### DISCUSSION TITLE
"Network Session 1" unowned output always present.. normal?!

### DISCUSSION THREAD

**Discussion by arshiacont** [orchetect/swift-midi#203] [Q&A] [answered]
Hello and thanks [again] for the great library!

When observing `MIDIManager`'s `observableEndpoints.outputs` there is _always_ an output called "Network Session 1" present in the list. I understand what Network sessions are, how to set them up and use them. But I am wondering if it is normal behavior to see this output "all the time" even if there is no MIDI-over-network anywhere nearby? In my limited experience Network sessions should appear once setup?!
If this is all normal, is there anyway to distinguish/filter between a `MIDIOutputEndpoint` originating from "Network" and other devices; or an active/non-active Endpoint?

_Motivation_: I am basically trying to distinguish between active/real outputs and those that are not! This one appears all the and somehow gets in the way of others. Didn't find a property to easily filter out. Sorry in advance if I missed something obvious.

**Comment by orchetect** [comment, marked as answer]
Yes I believe the network session endpoint persists regardless of connection status.

Without looking too deep, I could suggest matching against the reverse-DNS driver bundle ID to filter it out.

If you run the System Info example project and find the endpoint then click the Show All button it will give you the values of all its properties. That may give more hints.

Although unless you have a specific use case, it's best to allow all endpoints to be selectable by the user.

**Reply by arshiacont** [reply]
Thank you! That works.. for anyone coming across this, the `driverOwner` contains `"AppleMIDIRTPDriver"`  for network end points. The goal here being to avoid "auto-connect" to such devices. I agree that one should allow all endpoints to be selectable by the user.

Many thanks and great work again!
